action-server/covidflow/actions/question_answering_form.py

```diff
@@ -67,128 +63,109 @@
 }
 
 
-class QuestionAnsweringForm(FormAction):
+class ActionActivateFallbackQuestionAnswering(Action):
     def name(self) -> Text:
-
-        return FORM_NAME
+        return FALLBACK_ACTIVATE_ACTION_NAME
 
     async def run(
-        self, dispatcher, tracker, domain,
-    ):
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[EventType]:
         bind_logger(tracker)
-        return await super().run(dispatcher, tracker, domain)
-
-    ## override to play initial messages or prefill slots
-    async def _activate_if_required(
-        self,
-        dispatcher: CollectingDispatcher,
-        tracker: Tracker,
-        domain: Dict[Text, Any],
+
+        question = tracker.latest_message.get("text", "")
+
+        # Slot will be validated on form activation
+        return [SlotSet(QUESTION_SLOT, question)]
+
+
+class ActionAskActiveQuestion(Action):
+    def name(self) -> Text:
+        return ASK_QUESTION_ACTION_NAME
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
     ) -> List[EventType]:
-        if tracker.active_loop.get("name") == FORM_NAME:
-            return await super()._activate_if_required(dispatcher, tracker, domain)
+        bind_logger(tracker)
 
-        intent = get_intent(tracker)
+        events = []
 
-        # Fallback QA
-        if intent == FALLBACK_INTENT:
-            question = tracker.latest_message.get("text", "")
+        if tracker.get_slot(SKIP_QA_INTRO_SLOT) != True:
```

Comment:
Won't this trigger if the slot value is None? If that's intended maybe just use
`if not tracker.get_slot(SKIP_QA_INTRO_SLOT):`
or if not
`if tracker.get_slot(SKIP_QA_INTRO_SLOT) is False`
so we don't get weird bugs

Reply:
It is intended, but I prefer using checks to see if the value is True, as it should never be something with only truthy value. If there's an error and someone sets the value to `"false"`, it will still work. I could change it to `if not (tracker.get_slot(SKIP_QA_INTRO_SLOT) is True` to make it clearer though.